### ISSUE TITLE
Error in table alias keyword

### ISSUE THREAD

**Issue by fsblemos** [yajra/laravel-oci8#211]
### Summary of problem or feature request

Error when generating SQL for self relationships.

### Code snippet of problem

I have a model "Vendedor" which has a belongsTo self relationship called "superior":

```php
class Vendedor extends Model
{
    public function superior() {
        return $this->belongsTo(self::class, 'vendedor_id');
    }
}
``` 

When I run ```App\Vendedor::has('superior')->get()```, it generates the following SQL:

```sql
  select * 
    from vendedor 
   where exists (select * 
                   from vendedor as laravel_reserved_0 
                  where laravel_reserved_0.vendedor_id = vendedor.id)
```

However, the keyword "as" is used only in column alias, not in table alias (see [here](https://www.techonthenet.com/oracle/alias.php)). 

The generated error:
![image](https://cloud.githubusercontent.com/assets/14933454/19075491/7cb4c184-8a1b-11e6-8e4b-ea5c556c5404.png)

### System details
- Operating System: Windows 7
- PHP Version: 7.0
- Laravel Version: 5.2

**Comment by yajra** [comment]
Thanks for reporting, confirmed bug when using `self` as the related class but works well when using other class.

**Comment by yajra** [comment]
@fsblemos the issue is on the core framework. I made a PR for the fix. Let's hope it gets merged. Thanks!

**Comment by yajra** [comment]
@fsblemos, the PR was closed. Can you please check the PR I made if it works on your side? If yes, then I suggest you voice it out on the PR to be reconsidered? If it works and still make it thru the core, then I guess it will be known bug for the package. Thanks!

**Comment by fsblemos** [comment]
Hi @yajra, thank you! In some others databases the ```as``` for table alias is optional for both table and column. As I could see, only Oracle don't allow this keyword in table alias. It doesn't make more sense remove the ```as``` only for Oracle? Maybe it could be more safe than change the rule for all other possible dbs. What do you think?

**Comment by yajra** [comment]
@fsblemos that `as` keyword is one of the reasons why I created several packages and maintain another fork of some packages for them to work on Oracle. I was first hesitant to submit a PR to the community packages as it was only for Oracle and Laravel does not officially support it. However, after all the packages I went through, I just realized that it works on other databases too so I guess it is safer to remove `as` for broader compatibility. I think I also made some several attempts to submit a PR on other packages but was closed due to the fact that Oracle is not officially supported by Laravel.

A better example is my [datatables](https://github.com/yajra/laravel-datatables) package. It started as a fork of [Bllim](https://github.com/bllim/laravel4-datatables-package) because it doesn't work on Oracle because of `as` keyword when counting the records.

**Comment by yajra** [comment]
It seems like the SQLite grammar only processes table with `AS` aliasing. Sent another PR to also fix the SQLite grammar. Hopefully it will make it to the core this time.

**Comment by yajra** [comment]
@fsblemos as per the laravel core team via GrahamCampbell: 
> 5.2 is closed for changes. Please target 5.1 for fixes or 5.3.

Have re-submitted the PR to 5.3 branch. You may have to update your code to 5.3 if it get merged for this to be fixed.

**Comment by fsblemos** [comment]
Ok, I'll update the code. Thank you @yajra!

**Comment by yajra** [comment]
@fsblemos I just released a patch for Laravel 5.1, 5.2 & 5.3. Please update the package version and it should fix this issue. Thanks!
